core/shared/src/main/scala/zio/config/ConfigSourceModule.scala

```diff
@@ -17,20 +17,24 @@ trait ConfigSourceModule extends KeyValueModule {
 
   case class ConfigSource(
     sourceNames: Set[ConfigSource.ConfigSourceName],
-    access: ZManaged[Any, ReadError[K], PropertyTreePath[K] => UIO[ZIO[Any, ReadError[K], PropertyTree[K, V]]]],
+    access: ZManaged[
+      Any,
+      Nothing,
+      ZManaged[Any, ReadError[K], PropertyTreePath[K] => ZIO[Any, ReadError[K], PropertyTree[K, V]]]
+    ],
```

Comment:
Note that squashing it to a single `ZManaged` layer results in non-memoization behaviour, which actually makes sense considering the type returned by `.memoize` on `ZManaged`.